library/proc_macro/src/quote.rs

```diff
@@ -46,12 +47,12 @@ macro_rules! quote_ts {
 ///
 /// Note: supported tokens are a subset of the real `quote!`, but
 /// unquoting is different: instead of `$x`, this uses `(@ expr)`.
-macro_rules! quote {
+macro_rules! minimal_quote {
     () => { TokenStream::new() };
     ($($t:tt)*) => {
         [
-            $(TokenStream::from(quote_ts!($t)),)*
-        ].iter().cloned().collect::<TokenStream>()
+            $(ToTokens::into_token_stream(minimal_quote_ts!($t)),)*
+        ].into_iter().collect::<TokenStream>()
```

Comment:
I think this could be slightly optimized now with the new trait, and the `() => { TokenStream::new() };` pattern shouldn't even be needed anymore since `$(...)*` will still match if empty

```rust
macro_rules! minimal_quote {
    ($($t:tt)*) => {
        #[allow(unused_mut)] // in case the expansion is empty
        let mut ts = TokenStream::new();
        $(ToTokens::to_tokens(&minimal_quote_ts!($t), &mut ts); )*
        ts
    }
}
```

Reply:
Done.